File: objectiveai/objectiveai/src/functions/http.rs
use crate::{HttpClient, HttpError};

pub async fn list_functions(
    client: &HttpClient,
) -> Result<Vec<super::response::ListFunction>, HttpError> {
    client
        .send_unary(reqwest::Method::GET, "functions", None::<String>)
        .await
}

pub async fn get_function_usage(
    client: &HttpClient,
    fowner: &str,
    frepository: &str,
    fcommit: Option<&str>,
) -> Result<super::response::UsageFunction, HttpError> {
    let path = match fcommit {
        Some(fcommit) => {
            format!("functions/{}/{}/{}/usage", fowner, frepository, fcommit)
        }
        None => format!("functions/{}/{}/usage", fowner, frepository),
    };
    client
        .send_unary(reqwest::Method::GET, &path, None::<String>)
        .await
}

File: objectiveai-py/objectiveai/functions/http.py
"""HTTP functions for functions."""


async def list_functions(client):
    raise NotImplementedError


async def get_function(client, id):
    raise NotImplementedError


async def get_function_usage(client, id):
    raise NotImplementedError


async def list_function_profile_pairs(client):
    raise NotImplementedError


async def get_function_profile_pair_usage(client, id):
    raise NotImplementedError
